source/grid/grid_in.cc

```diff
@@ -2760,9 +2792,542 @@ GridIn<dim, spacedim>::parse_format (const std::string &format_name)
 template <int dim, int spacedim>
 std::string GridIn<dim, spacedim>::get_format_names ()
 {
-  return "dbmesh|msh|unv|vtk|ucd|xda|netcdf|tecplot";
+  return "dbmesh|msh|unv|vtk|ucd|abaqus|xda|netcdf|tecplot";
 }
 
+namespace
+{
+  template <int dim>
+  class Abaqus_to_UCD
+    {
+    public:
+      Abaqus_to_UCD (void);
+      ~Abaqus_to_UCD (void);
+
+      bool read_in_abaqus (std::istream &in);
+      bool write_out_avs_ucd (std::ostream &out) const;
+
+    private:
+      const double tolerance;
+
+      std::vector<double> get_global_node_numbers (const int face_cell_no,
+                                                   const int face_cell_face_no) const;
+
+      // NL: Stored as [ global node-id (int), x-coord, y-coord, z-coord ]
+      std::vector< std::vector<double> >  node_list;
+      // CL: Stored as [ material-id (int), node1, node2, node3, node4, node5, node6, node7, node8 ]
+      std::vector< std::vector<double> >  cell_list;
+      // FL: Stored as [ sideset-id (int), node1, node2, node3, node4 ]
+      std::vector< std::vector<double> >  face_list;
+      // ELSET: Stored as [ (std::string) elset_name = (std::vector) of cells numbers]
+      std::map< std::string, std::vector<int> > elsets_list;
+    };
+
+  template <int dim>
+  Abaqus_to_UCD<dim>::Abaqus_to_UCD (void)
+    : tolerance (5e-16) // Used to offset Cubit tolerance error when outputting value close to zero
+  {
+    AssertThrow(dim==2 || dim==3, ExcNotImplemented());
+  }
+
+  template <int dim>
+  Abaqus_to_UCD<dim>::~Abaqus_to_UCD (void)
+  { }
+
+  // Convert from a string to some other data type
+  // Refrence: http://www.codeguru.com/forum/showthread.php?t=231054
```

Comment:
@tjhei @bangerth Does it make sense adding either of the functions `from_string` or `extract_int` to the `Utilities` namespace?